pkgs/by-name/ka/kasmvnc/package.nix

```diff
@@ -0,0 +1,126 @@
+{
+  stdenv,
+  fetchurl,
+  dpkg,
+  lib,
+  perl,
+  perlPackages,
+  libpng,
+  libunwind,
+  pixman,
+  makeWrapper,
+  mesa,
+  libXfont2,
+  libXau,
+  libxshmfence,
+  libXdmcp,
+  libwebp,
+  libxcrypt-legacy,
+  freetype,
+  libbsd,
+  kasmvnc-xvnc
+}:
+let
+  rpathLibs = [
+    freetype
+    pixman
+    libpng
+    libunwind
+    libXfont2
+    libXau
+    libxshmfence
+    libXdmcp
+    libwebp
+    libxcrypt-legacy
+    libbsd
+    mesa
+  ];
+
+  perlLibs = with perlPackages; [
+    Switch
+    ListMoreUtils
+    ExporterTiny
+    TryTiny
+    DateTime
+    namespaceautoclean
+    namespaceclean
+    BHooksEndOfScope
+    ModuleImplementation
+    ModuleRuntime
+    SubExporterProgressive
+    PackageStash
+    SubIdentify
+    Specio
+    MROCompat
+    RoleTiny
+    EvalClosure
+    DevelStackTrace
+    ParamsValidationCompiler
+    ExceptionClass
+    ClassDataInheritable
+    DateTimeLocale
+    FileShareDir
+    ClassInspector
+    DateTimeTimeZone
+    ClassSingleton
+    YAMLTiny
+    HashMergeSimple
+  ];
+
+  mkPerlLibraryPath = ppkgs:
+    builtins.concatStringsSep ":" (
+      map (pkg: "${pkg}/lib/perl5/site_perl") ppkgs
+    );
+in
+stdenv.mkDerivation rec {
+  pname = "kasmvnc";
+  # These two are version linked because they have to be built from the same source, though are different outputs.
+  version = kasmvnc-xvnc.version;
+
+  # We do not build from source because kasmvnc only provides a docker build env. This also does not give
+  # us static linked builds. Thus it doesn't really matter.
```

Comment:
From-source builds are always preferred. Please at least try to do it, it matters rather quite a lot and might even be simpler.

Docker envs are usually only for providing the deps which we can do just fine in Nix.

If it really is too much of a pain to build from source, a binary build is acceptable but not preferred.

Reply:
You can probably ignore all the docker stuff and just follow this:

https://github.com/kasmtech/KasmVNC/blob/master/BUILDING.txt